cpp/src/generic_players/main/UnitTests.cpp

```diff
@@ -0,0 +1,146 @@
+#include <core/concepts/Game.hpp>
+#include <core/GameTypes.hpp>
+#include <core/BasicTypes.hpp>
+#include <games/tictactoe/Game.hpp>
+#include <generic_players/MctsPlayer.hpp>
+#include <mcts/Manager.hpp>
+#include <mcts/ManagerParams.hpp>
+#include <mcts/SearchLog.hpp>
+#include <util/EigenUtil.hpp>
+#include <util/BoostUtil.hpp>
+
+#include <gtest/gtest.h>
+
+#include <fstream>
+
+namespace generic {
+
+template <core::concepts::Game Game>
+class MctsPlayerTest : public ::testing::Test {
+ protected:
+  using Manager = mcts::Manager<Game>;
+  using ManagerParams = mcts::ManagerParams<Game>;
+  using MctsPlayer = generic::MctsPlayer<Game>;
+  using MctsPlayerParams = MctsPlayer::Params;
+  using SearchResults = Game::Types::SearchResults;
+  using SearchLog = mcts::SearchLog<Game>;
+  using PolicyTensor = Game::Types::PolicyTensor;
+  using StateHistory = Game::StateHistory;
+  using State = Game::State;
+  using ActionMask = Game::Types::ActionMask;
+  using Service = mcts::NNEvaluationServiceBase<Game>;
+  using Rules = Game::Rules;
+
+ public:
+  MctsPlayerTest() : manager_params_(create_manager_params()), player_params_(mcts::kCompetitive) {}
+
+  ManagerParams create_manager_params() {
+    ManagerParams params(mcts::kCompetitive);
+    params.no_model = true;
+    return params;
+  }
+
+  void init(Service* service = nullptr) {
+    mcts_manager_ = new Manager(manager_params_, service);
+    const mcts::SharedData<Game>* shared_data = mcts_manager_->shared_data();
+    search_log_ = new mcts::SearchLog<Game>(shared_data);
+    mcts_manager_->set_post_visit_func([&] { search_log_->update(); });
+    mcts_manager_->start();
+    mcts_manager_->set_player_data(mcts_manager_->shared_data());
+    mcts_player_ = new MctsPlayer(player_params_, mcts_manager_);
```

Comment:
It would be better to add an optional third parameter to the MctsPlayer, `bool owns_manager = false`. 

Then, you can remove the `start()` and `set_player_data()` calls here.

This makes it less likely that this test logic will break as the `MctsPlayer` code evolves in the future.